src/Form.js

```diff
@@ -0,0 +1,199 @@
+import React, { useState, useRef } from "react";
+import "./Form.css";
+const Form = () => {
+  const [formData, setFormData] = useState({
+    loginId: "",
+    email: "",
+    name: "",
+    timezone: "",
+    homePage: "",
+    aboutMe: "",
+    receiveNotifications: false
+  });
+  const selectRef = useRef(null);
+  const [errors, setErrors] = useState({});
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormData({ ...formData, [name]: value });
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const validationErrors = validateFormData(formData);
+    if (Object.keys(validationErrors).length > 0) {
+      setErrors(validationErrors);
+    } else {
+      setFormData({
+        loginId: "",
+        email: "",
+        name: "",
+        timezone: "",
+        homePage: "",
+        aboutMe: "",
+        receiveNotifications: false
+      });
+      setErrors({});
+    }
+  };
+  const validateFormData = (formData) => {
+    const errors = {};
+
+    if (!formData.loginId) {
+      errors.loginId = "Login ID is required";
+    }
+    if (!formData.email) {
+      errors.email = "Email is required";
+    } else if (!/\S+@\S+\.\S+/.test(formData.email)) {
+      errors.email = "Invalid email format";
+    }
+    if (!formData.name) {
+      errors.name = "Name is required";
+    }
+    if (document.querySelector("select").value === "") {
+      errors.timezone = "Timezone is required";
+    }
+    if (!formData.homePage) {
+      errors.homePage = "Home page is required";
+    } else if (
+      !/^(http|https):\/\/[a-zA-Z0-9]+([\-\.]{1}[a-zA-Z0-9]+)*\.[a-zA-Z]{2,20}(:[0-9]{1,5})?(\/.*)?$/.test(
+        formData.homePage
+      )
+    ) {
+      errors.homePage = "Invalid URL format";
+    }
+    if (formData.aboutMe.length > 0 && formData.aboutMe.length < 50) {
+      errors.aboutMe = "About Me must be at least 50 characters";
+    }
+    if (!formData.receiveNotifications) {
+      errors.receiveNotifications = "You must accept to receive notifications";
+    }
```

Comment:
create a separate file for these utils. which will give you the errors.. let's connect on call regarding this change.